### ISSUE TITLE
Block switcher a11y improvements

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#695]
Pretty much like in #527, see #525 

The block switcher is made of a toggle button and a menu, exactly as the Inserter. It needs a very similar accessibility treatment, adding ARIA roles, states, and properties.

Currently, the toggle button has no label:

![block switcher](https://cloud.githubusercontent.com/assets/1682452/25773162/0ddd607a-3279-11e7-840f-04900f9bfe74.jpg)

The menu needs arrow navigation, as done for the Inserter, see #578. Will open a separate issue.
